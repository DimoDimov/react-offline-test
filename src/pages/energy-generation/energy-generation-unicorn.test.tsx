import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
jest.mock('@/shared/http-services/data-api.service');
import EnergyGenerationUnicorn from './energy-generation-unicorn.component';

describe('<EnergyGenerationUnicorn />', () => {
    let container = null;
    beforeEach(() => {
        // setup a DOM element as a render target
        container = document.createElement('div');
        document.body.appendChild(container);
        act(() => {
            render(<EnergyGenerationUnicorn />, container);
        });
    });

    afterEach(() => {
        // cleanup on exiting
        unmountComponentAtNode(container);
        container.remove();
        container = null;
    });

    it('should render and match snapshot', () => {
        expect(pretty(container.innerHTML)).toMatchSnapshot();
    });

    it('should display a unicorn', () => {
        expect(container.querySelectorAll('.unicorn-image').length).toBe(1);
    });
});
