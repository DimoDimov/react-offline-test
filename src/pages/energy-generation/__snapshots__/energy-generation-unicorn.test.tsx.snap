// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EnergyGenerationUnicorn /> should render and match snapshot 1`] = `
"<div class=\\"container unicorn-container mt-5\\">
  <h4>And of course giant unicorn's rainbow puke!!</h4>
  <div class=\\"col-xs-6\\"><span class=\\"cursor-pointer\\"><img class=\\"unicorn-image\\" src=\\"\\" alt=\\"puking unicor\\"></span></div>
</div>"
`;
